kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationFilter.java

```
@@ -109,7 +108,18 @@ public CompletionStage<RequestFilterResult> onSaslAuthenticateRequest(short apiV
             return context.forwardRequest(header, request);
         }
         else {
-            return authenticate(request.authBytes())
+            var server = saslServer;
+            if (server == null) {
```

i think failing early in this case makes the code easier to reason about.